### PR TITLE
test(webhook): add unit tests for CappMutator

### PR DESCRIPTION
**Pull request by sbahar619**
Covers mutation, annotation injection, and default resource assignment including edge cases. Shared test fixtures extracted to a single file used by both validator and mutator suites.

#691

### CODE REVIEW

File: internal/webhook/rcs/v1alpha1/mutator_test.go [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+func newCappMutator(t *testing.T, scheme *runtime.Scheme) *CappMutator {
+	t.Helper()
+
+	cfg := newCappConfig()
+	cfg.Spec.DefaultResources = newDefaultResources()
+
```
Comment (elmatuso) [inline internal/webhook/rcs/v1alpha1/mutator_test.go:25]: DefaultResources is always populated here. Consider adding a test case where DefaultResources is zero-valued to verify the mutator doesn't inject empty requests/limits (or that it correctly skips mutation).

File: internal/webhook/rcs/v1alpha1/mutator_test.go [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+	fakeClient := fake.NewClientBuilder().
+		WithScheme(scheme).
+		WithObjects(cfg).
```
Comment (elmatuso) [inline internal/webhook/rcs/v1alpha1/mutator_test.go:28]: Consider adding a test where the fake client has no CappConfig object (i.e., drop WithObjects(cfg)). This would verify the mutator degrades gracefully when the config is missing and help prevent null pointer panics in production.

File: internal/webhook/rcs/v1alpha1/mutator_test.go [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+		capp.Name = cappName
+		capp.Namespace = nsName
+		capp.Spec.ConfigurationSpec.Template.Spec.Containers = []corev1.Container{
+			{Name: testContainerName},
+		}
+
+		raw, err := json.Marshal(capp)
+		require.NoError(t, err)
+
+		req := admission.Request{
```
Comment (elmatuso) [inline internal/webhook/rcs/v1alpha1/mutator_test.go:110]: This only tests Create. Since the annotation is named lastUpdatedBy, I'd expect it to also be set on Update operations. Adding a test with admissionv1.Update (plus OldObject) would document and verify that behavior.

File: internal/webhook/rcs/v1alpha1/mutator_test.go [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+
+		first := capp.Spec.ConfigurationSpec.Template.Spec.Containers[0]
+		require.Equal(t, resource.MustParse("1"), first.Resources.Requests[corev1.ResourceCPU])
+		require.Equal(t, resource.MustParse("2Gi"), first.Resources.Limits[corev1.ResourceMemory])
+
+		second := capp.Spec.ConfigurationSpec.Template.Spec.Containers[1]
+		require.Equal(t, resource.MustParse("100m"), second.Resources.Requests[corev1.ResourceCPU])
+		require.Equal(t, resource.MustParse("512Mi"), second.Resources.Limits[corev1.ResourceMemory])
+	})
+}
```
Comment (elmatuso) [inline internal/webhook/rcs/v1alpha1/mutator_test.go:277]: Consider adding a TestMutateResources subtest for an empty containers slice ([]corev1.Container{}) to document that mutateResources is a no-op in that case and prevent future regressions.

File: internal/webhook/rcs/v1alpha1/mutator_test.go (hunk 1 above)
Comment (sbahar619) [reply internal/webhook/rcs/v1alpha1/mutator_test.go:25]: You're right there's no explicit test for the zero-valued path.
The current implementation is safe (nil-map iteration is a no-op in Go), but I'll add a sub-test to lock that in and guard against future refactors.

File: internal/webhook/rcs/v1alpha1/mutator_test.go (hunk 2 above)
Comment (sbahar619) [reply internal/webhook/rcs/v1alpha1/mutator_test.go:28]: Added a test confirming Handle returns 500 when CappConfig is missing from the cluster — covers the GetCappConfig error branch.

File: internal/webhook/rcs/v1alpha1/mutator_test.go (hunk 3 above)
Comment (sbahar619) [reply internal/webhook/rcs/v1alpha1/mutator_test.go:110]: The mutation path doesn't branch on operation, Create and Update hit identical logic (Operation isn't checked).
Adding an Update test would duplicate the existing coverage without new signal.
The webhook registration (verbs=create;update) already declares it handles both.

File: internal/webhook/rcs/v1alpha1/mutator_test.go (hunk 4 above)
Comment (sbahar619) [reply internal/webhook/rcs/v1alpha1/mutator_test.go:277]: The function ranges over containers, an empty slice naturally results in zero iterations with nothing to regress on.
I'd prefer not to add a test for basic range semantics here.

**Review by elmatuso — APPROVED** [review]
/lgtm
